modules/workergroup/nodepools.tf

```diff
@@ -0,0 +1,153 @@
+# Copyright 2017, 2022, Oracle Corporation and/or affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl
+
+# Dynamic resource block for Node Pool groups defined in worker_groups
+resource "oci_containerengine_node_pool" "nodepools" {
+  for_each = {
```

Comment:
Can you please elaborate on the algorithm behind this?

Reply:
Will clarify in comments also: this applies the node_pool resource for elements in worker_groups defined with that mode only, that are also enabled w/ "create" (pending rename based on feedback).